cmake/compiler_flag_sets/cortex-m4-fpv4-sp-d16-nosys.cmake

```diff
@@ -0,0 +1,97 @@
+#
+# Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# C, CXX, LD, and AS flags for native targets.
+#
+
+set(MCU_ARCH cortex-m4)
+set(MCU_FLOAT_ABI hard)
+set(MCU_FLOAT_FPU fpv4-sp-d16)
+set(MCU_CORTEX_MATH_LIB "arm_cortexM4lf_math")
+
+set(DEFINITIONS_SET 
+                "-DARM_MATH_CM4=true"
+                "-D__FPU_PRESENT=1"
+)
+
+set(C_FLAG_SET  "-mcpu=${MCU_ARCH}"
+                "-mthumb"
+                "-mno-thumb-interwork"
+                "-mfpu=${MCU_FLOAT_FPU}"
+                "-mfloat-abi=${MCU_FLOAT_ABI}"
+                "-ffunction-sections"
+                "-fdata-sections"
+                "-fno-strict-aliasing"
+)
+
+#
+# Debug flags for C and C++
+#
+list(APPEND C_FLAG_SET
+                "-Og"
+                "-DDEBUG"
+                "-ggdb"
+)
+
+#
+# Warnings and other diagnostics for C and C++
+#
+list(APPEND C_FLAG_SET
+                "-pedantic"
+                "-Wall"
+                "-Wextra"
+                "-Werror"
+                "-Wfloat-equal"
+                "-Wabi"
+                "-Wunused-parameter"
+                "-Wunused-variable"
+                "-Wunused-value"
+                "-Wno-missing-field-initializers"
```

Comment:
Why?

Reply:
The vendor code and gtest are written pretty loosely. I'm not as concerned about this given that we cover our code with stricter flags for the native builds.

Reply:
I am questioning the necessity of `-Wno-missing-field-initializers`. Is there any third-party code which has missing field initializers, or is this merely a copy-pasting mistake?

Reply:
Ah. Yes, we need it to build gtest for ontarget but the native one is a copy and paste error. Will remove.